### PR TITLE
Issue templates proposal 

### PR DESCRIPTION
**Pull request by spdegabrielle**
EDIT: reason - text from post to racket-dev

> I was recently involved in a discussion on the slack that mentioned the challenge of transitioning racketeers from users who log issues(“complain and someone will fix it”)* or offer opinions(without the time or interest to make a contribution)* into contributors.
> 
> While racket has many great contributors, this is probably not uncommon for OSS communities, it made me think about ways to 'nudge' users towards becoming contributors. 
> 
> The following text in issue templates is a small attempt to nudge users to contribute, while being explicit about what is useful in bug and feature reports respectively.

--
shamelessly copied from Pharo 

see in action
https://github.com/spdegabrielle/clash/issues/new/choose

### CODE REVIEW

**Comment by spdegabrielle** [comment]
hey @samth @mflatt what do you think?

**Comment by spdegabrielle** [comment]
hey @samth @mflatt what do you think?

**Comment by mflatt** [comment]
This generally looks like a good idea to me. Can we check with someone at Pharo to make sure it's ok to copy their template?

I'd leave out the "development cost" part for a bug report, because I'd rather hear about bugs independent of how difficult someone thinks it would be to fix. I'm less sure about that question for an issue; it's a fair question, but I think everything is harder than someone thinks :), so I'm not sure it's helpful to ask them to estimate.

**Comment by samth** [comment]
I like the idea of multiple templates. I think we should be a little more general in the description. For example, the TR issue template here: https://github.com/racket/typed-racket/issues/new is much less specific. We might want a little more information than that, but not much.

I agree with Matthew that estimating development cost is unlikely to be helpful; seemingly very small bugs can be very difficult to fix, and the reverse as well.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+name: Feature request
+about: Suggest an idea to discuss
+title: "[Feature Request]"
+labels: feature-request
+assignees: ''
+
+---
+
+_A new feature is some behavior that is not existing already. If what you describe is a slight variation of something already existing please use a enhancement request._
```
Comment (sorawee) [inline .github/ISSUE_TEMPLATE/feature_request.md:10]: There's no more enhancement request. This needs to be updated.

**Comment by spdegabrielle** [comment]
@mflatt 

> This generally looks like a good idea to me. Can we check with someone at Pharo to make sure it's ok to copy their template?

I've asked as an issue against their main repo https://github.com/pharo-project/pharo/issues/6708

> I'd leave out the "development cost" part for a bug report, because I'd rather hear about bugs independent of how difficult someone thinks it would be to fix. I'm less sure about that question for an issue; it's a fair question, but I think everything is harder than someone thinks :), so I'm not sure it's helpful to ask them to estimate.

Done

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 1 above)
Comment (spdegabrielle) [reply .github/ISSUE_TEMPLATE/feature_request.md]: done

**Comment by spdegabrielle** [comment]
@samth 

I'm thinking issues logged against this repo will probably need to accommodate DrRacket and other components of the racket ecosystem (website, mailing lists, packages catalog, etc.) , hence screenshots, steps to reproduce.

Is this better?  (or worse)
```
[please fill in as much as appropriate]

**Describe the bug**
 A clear and concise description of what the bug is.

**Racket Version**
[e.g. 7.7]
if you are not using the current release you issue may have already been resolved.

**To Reproduce**
What program did you run?
```
[insert code here]
```

Steps to reproduce the behavior?
 1. Go to '...'
 2. Click on '....'
 3. Scroll down to '....'
 4. See error

 **Expected behavior**
What should have happened? 

 **Error message/screenhots**
 If you got an error message, please include it here.
 If applicable, add screenshots to help explain your problem.

 **Platform Version information:**
Windows / Macos/ Linux?
Version?
32/64 bit, x86/ARM?

 **Additional context**
 Add any other context about the problem here.
```

**Comment by spdegabrielle** [comment]
@mflatt @samth I left "Development cost" in the feature request template

NB I haven't applied these changes to the demo as they are only changing the text that gets inserted in the template, not how GitHub presents the template

**Comment by mfelleisen** [comment]
FWIW, the Racket people are _not_ a sw dev shop. (We just pretend to be one on occasion.) I think that would be my philosophical objection to "cost estimates".  (This means that academic obligations distract a lot more than distractions in a normal sw dev shop or even a research lab such as INRIA or MSR.)

**Comment by spdegabrielle** [comment]
@mfelleisen 

The intent was to  remind the submitter that this is a volunteer driven project.
```
**Expected development cost**
 As you know, nothing happens magically, so do you have an idea of the development effort required? 
 In addition, are you willing to help because after all Racket is yours too?
```
Maybe a better approach is something like this ?

```
**Expected development effort**
Racket is yours too! What is the development effort required and how can you contribute?
Has this been implemented or a fork or as a package?
- If so have documentation and tests been created?
Does the code follow the Racket coding standard/style guide/contributor guidelines?
```

**Comment by spdegabrielle** [comment]
@mflatt

> Can we check with someone at Pharo to make sure it's ok to copy their template?

Permission granted with thanks to @MarcusDenker  https://github.com/pharo-project/pharo/issues/6708#issuecomment-650183258

**Comment by spdegabrielle** [comment]
Hi @mfelleisen 

I've come up with a shorter version of the **Expected development cost** section that removed mention of **cost**.

```
**Expected development effort**
Racket is yours too!
What is the development effort required and how can you contribute?
```

What do you think?

Kind regards

Stephen

**Comment by spdegabrielle** [comment]
@samth I've changed the bug report template to more closely match the TR bug report.

Please let me know what you think

Kind regards, 

Stephen

**Comment by samth** [comment]
I think that instead of "Expected development cost", we should ask people if they want to help develop the feature. For example:

```
***Do you want to contribute to this feature***
If you're interested in working on the development of this feature, say so here. 
```

**Comment by spdegabrielle** [comment]
@samth done!

**Comment by spdegabrielle** [comment]
Hi All, 

I hope I've addressed all the concerns. Please let me know if there is any further barriers to adoption.

Kind regards, 

Stephen

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+title: "[Feature Request]"
+labels: feature-request
+assignees: ''
+
+---
+
+_A new feature is some behavior that is not existing already._
+
+**Is your feature request related to a problem? Please describe.**
+A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
```
Comment (samth) [inline .github/ISSUE_TEMPLATE/feature_request.md:13]: Can we take out the phrase "clear and concise" in both places?

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+name: Bug report
+about: Create a report to help us improve Racket 
+title: ''
+labels: bug
```
Comment (samth) [inline .github/ISSUE_TEMPLATE/bug-report.md:5]: I don't think this needs a label by default.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+name: Bug report
+about: Create a report to help us improve Racket
```
Comment (samth) [inline .github/ISSUE_TEMPLATE/bug-report.md:3]:
```suggestion
about: Report a problem with Racket or Racket's tools
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+---
+name: Feature request
+about: Suggest an idea to discuss
+title: "[Feature Request]"
+labels: feature-request
+assignees: ''
+
+---
+
+_A new feature is some behavior that is not existing already._
```
Comment (samth) [inline .github/ISSUE_TEMPLATE/feature_request.md:10]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 3 above)
Comment (sorawee) [reply .github/ISSUE_TEMPLATE/bug-report.md:5]: I kinda disagree. Label makes it possible to filter issues. I think we should keep it.

**Comment by sorawee** [comment]
`ISSUE_TEMPLATE_DRAFT` probably should be deleted, too.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 3 above)
Comment (samth) [reply .github/ISSUE_TEMPLATE/bug-report.md:5]: Labels are great, but I'm not sure "bug" is a useful one.

Comment (sorawee) [reply .github/ISSUE_TEMPLATE/bug-report.md:5]: IIUC, it's to distinguish bugs from feature requests.

Granted, Racket seems to have bug issues far more than feature request issues.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (spdegabrielle) [reply .github/ISSUE_TEMPLATE/feature_request.md]: removed from all three.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 3 above)
Comment (spdegabrielle) [reply .github/ISSUE_TEMPLATE/bug-report.md]: I've removed the label - everything is a bug if it is not a feature request. Note the current cohort of bug reports don't have a label so this is consistent.

**Comment by spdegabrielle** [comment]
@sorawee 
> `ISSUE_TEMPLATE_DRAFT` probably should be deleted, too.

Done

**Comment by spdegabrielle** [comment]
@samth 
> Can we make the formatting consistent between the two templates? I'm indifferent as to which is better.

Done!

**Comment by spdegabrielle** [comment]
Hi All,

I hope I've addressed all the concerns. Please let me know if there is any further barriers to adoption.

Kind regards,

Stephen

**Comment by spdegabrielle** [comment]
Hi All, any chance of a LGTM?

**Comment by spdegabrielle** [comment]
Hi I've updated the description of this PR 

Reason - text from post to racket-dev

> I was recently involved in a discussion on the slack that mentioned the challenge of transitioning racketeers from users who log issues(“complain and someone will fix it”)* or offer opinions(without the time or interest to make a contribution)* into contributors.
> 
> While racket has many great contributors, this is probably not uncommon for OSS communities, it made me think about ways to 'nudge' users towards becoming contributors. 
> 
> The following text in issue templates is a small attempt to nudge users to contribute, while being explicit about what is useful in bug and feature reports respectively.

**Comment by mflatt** [comment]
LGTM. Thanks for all the improvements!

Just to be sure, I'll defer to @samth to merge (though it may take an extra day).

**Pull request merged**